### PR TITLE
Pre-materialize tag values for ENV interceptor at initialization to reduce CPU usage per-span

### PR DESCRIPTION
**Pull request by splittingred**
The ENV interceptor is currently collecting and parsing values from ENV per-span. This changes it to only collect and parse the values during initialization, and only add the tags to the span in the actual span block.

----

@bigcommerce/platform-engineering @jmwiese @pedelman @mattolson

### CODE REVIEW

File: spec/bigcommerce/lightstep/interceptors/env_spec.rb [hunk 1]
```diff
@@ -44,6 +44,14 @@
           expect(span).to receive(:set_tag).with('provider.datacenter', 'us-central1-a').once.ordered
           subject
         end
+
+        it 'should only collect the tags once even with multiple calls' do
+          expect(env).to receive(:fetch).with('GIT_SHA', nil).once.ordered
+          expect(env).to receive(:fetch).with('DATACENTER', nil).once.ordered
+          subject
+          subject
```
Comment (pedelman) [inline spec/bigcommerce/lightstep/interceptors/env_spec.rb]: Isnt subject memoized within a test?

Comment (splittingred) [reply spec/bigcommerce/lightstep/interceptors/env_spec.rb]: 🤔 I didn't think so, but worth adjusting to be entirely sure.

Comment (pedelman) [reply spec/bigcommerce/lightstep/interceptors/env_spec.rb]: https://relishapp.com/rspec/rspec-core/docs/subject/explicit-subject#the-%60subject%60-is-memoized-within-an-example-but-not-across-examples

Comment (splittingred) [reply spec/bigcommerce/lightstep/interceptors/env_spec.rb]: TIL! Thanks!

**Pull request merged**